include/nbl/video/utilities/IUtilities.h

```diff
@@ -104,6 +138,7 @@ class IUtilities : public core::IReferenceCounted
             submit.pWaitSemaphores = semaphoresToWaitBeforeExecution;
             submit.pWaitDstStageMask = stagesToWaitForPerSemaphore;
             queue->submit(1u,&submit,fence);
+            m_device->waitForFences(1u, &fence, false, 9999999999ull);
```

Comment:
fix the above to do filling directly from the staging buffer without creating an intermediate, then you'll get the opportunity to add the fence and the commandbuffer to the staging buffer's zombie list during a `multi_free`